src/json/from_json.cr

```diff
@@ -65,6 +65,57 @@ def Deque.from_json(string_or_io) : Nil
   end
 end
 
+module Iterator(T)
+  # Reads the content of a JSON array into an iterator in a lazy way.
+  # With this method it should be possible to process a huge JSON array, without
+  # the requirement that the whole array fits into memory.
+  #
+  # The following example produces a huge file, uses a lot of CPU but should not require much memory.
```

Comment:
Just wondering does this actually use "a lot of CPU"? It might take some time, sure. But I wouldn't expect it to be a particularly CPU-heavy task.